### DISCUSSION TITLE
New special block wich acts like a line or lines of code

### DISCUSSION THREAD

**Discussion by vanczakp** [pybricks/support#1535] [Ideas]
Hello! I really like the block like "coding" but some advanced feature is not available at block coding environment neighter switching between code and blocks works. Would it be possible to add a block which in it I can put a code and acts like a new line in the code? Both setup and program block would be good. Thanks!

**Comment by laurensvalk** [comment]
You can write your code in a Python module and import it into your block program (see the external task blocks).

Would this work for your use case?

**Reply by vanczakp** [reply]
It would be for block coding. I have never learnt coding, but with block I can do everything I want, as complex as I want but with the limitation of available blocks. With this special block we would be able to make the program more complex while keeping the block environment. Of course I can copy the generated code into code environment with getting rid of the first line but each update become more complex.

**Reply by laurensvalk** [reply]
Have you checked the external task blocks? I think they do what you need.

**Reply by vanczakp** [reply]
I have seen it and I see what it means but that requires an another "program" which contains that code. Would it be possible to add a new block so no other programs required? :)

**Reply by laurensvalk** [reply]
It is probably possible but we try to strike the right balance.

If we had a block with a text field for Python code, I think we'd rapidly see projects with a few blocks and giant blobs of Python code inside. This would be a nightmare to maintain and format correctly, preserving indentation, and so on.

Having it in a dedicated Python file solves all of these problems. It also keeps the code clean, and requires maybe just a few extra seconds to set up :)

**Reply by vanczakp** [reply]
I get your point and agree with you!

**Reply by creadvty** [reply]
Hi a text code block would be really helpful for learning text based coding. To illustrate, this video about vex switch blocks: https://youtu.be/oZ1ozJp4DMw?si=FL_Bv4ye7VHvGSB6

**Reply by laurensvalk** [reply]
You can call Python code directly from the blocks. Have you tried the `import` block?
